lib/utils/url.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+if [ -z "${BPKG_URL}" ]; then
+  BPKG_URL=1
+
+  ## Collection of shared bpkg functions
+
+  bpkg_resolve_uri_default() {
+    local git_remote=$1
+    local user=$2
+    local name=$3
+    local version=$4
+
+    if [[ "$git_remote" == https://gitlab*  ]]; then
```

Comment:
Let's say I have a popular package at `https://github.com/bash-bastion/bash-object`. I think this code allows a malicious actor to register the `bash-bastion` organization at GitLab, and "redirect" the downloads to themselves? In that case, should we allow the possibility for the domain to be specified verbosely (ex. `gitlab.com/bash-bastion/bash-object`), or would that be out of scope?

Reply:
A good point but if we don't allow git remote control we also have to disable it from `~/.bpkgrc` by not allowing to override `BPKG_REMOTES` and `BPKG_GIT_REMOTES`

Reply:
I'm not sure I quite understand? It is true that we don't have to allow overriding `BPKG_REMOTES` and `BPKG_GIT_REMOTES`, which will mean that this "exploitation" possibility won't exist. If we allow to control the Git remote, then there should probably be a way to mitigate this issue. I read the diff again, and it seems like to fix this, a different part of the codebase will need to be modified.